### PR TITLE
Decompile RVL/PAD

### PR DESCRIPTION
**Pull request by wait-wtf**
- [x] PAD_ClampCircle
- [x] PADClampCircle
- [x] PADClampCircle2
- [x] PAD_UpdateOrigin
- [x] PADOriginCallback
- [x] PADOriginUpdateCallback
- [x] PADProbeCallback
- [x] PADTypeAndStatusCallback
- [x] PADReceiveCheckCallback
- [x] PADReset
- [x] PADRecalibrate
- [x] PADInit
- [x] PADRead
- [x] PADControlMotor
- [x] SPEC0_MakeStatus
- [x] SPEC1_MakeStatus
- [x] SPEC2_MakeStatus
- [x] PAD_OnReset
- [x] PAD_SamplingHandler
- [x] __PADDisableRecalibration

### CODE REVIEW

**Comment by wait-wtf** [comment]
@riidefi Could you help me out with PADRead()? I'm having an issue with inlining that I can't figure out myself. Might have another look on Saturday.

**Comment by riidefi** [comment]
> @riidefi Could you help me out with PADRead()? I'm having an issue with inlining that I can't figure out myself. Might have another look on Saturday.
I'll take a look tomorrow, let's make sure we're using the proper compiler version, too.

**Comment by wait-wtf** [comment]
PADRead() is fixed now by using the `__abs()` compiler intrinsic. PAD_OnReset remains broken.

**Comment by wait-wtf** [comment]
PAD is now fully decompiled and working.

File: source/rvl/pad/rvlPad.c [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+  u32 unk_0A;
+} OSResetFunctionInfo;
+void OSSetCurrentContext(OSContext*);
+void OSClearContext(OSContext*);
+void OSRegisterVersion(const char*);
+s64 OSGetTime();
+void OSRegisterResetFunction(OSResetFunctionInfo* info);
+// Broadway / IOS global locations: https://wiibrew.org/wiki/Memory_Map
+u8 oslow_30e3 : (0x800030e3);
+u16 oslow_30e0 : (0x800030e0);
```
Comment (riidefi) [inline source/rvl/pad/rvlPad.c:26]: memest/memcpy -> string.h
OS* can go in "osMisc.h" or something now, we can sort them later.

**Review by riidefi — CHANGES_REQUESTED** [review]

**Pull request merged**